### PR TITLE
Us51 i18 n manage rules

### PR DESCRIPTION
**Pull request by AmineChalouati**

### CODE REVIEW

**Review by mkrout — DISMISSED** [review]

File: portlets/src/main/frontend/src/apps/components/rule/ManageRules.vue [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
-          :show="dismissCountDown"
-          dismissible
-          fade
-          variant="danger"
-          @dismiss-count-down="countDownChanged"><i class="uiIconError"></i>  An error happen when adding a rule</b-alert>
+
+        <div class="alert alert-success" v-if="isadded || addSuccess" v-on:="closeAlert()">
+
+            <i class="uiIconSuccess"></i>
+            Rule {{updateMessage}} successfully
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/ManageRules.vue:8]: not i18ned ?

File: portlets/src/main/frontend/src/apps/components/rule/ManageRules.vue [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+        </div>
+
+
+        <alert @dismiss-count-down="countDownChanged"
+               class="alert alert-danger"
+               :show="dismissCountDown"
+               dismissible
+               fade
+               variant="danger"
+               v-if="addError"><i class="uiIconError"></i> An error happen when adding a rule
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/ManageRules.vue:18]: not i18ned ?

File: portlets/src/main/frontend/src/apps/components/rule/RuleList.vue [hunk 3]
```diff
@@ -2,39 +2,118 @@
     <b-container fluid>
         <b-row>
             <b-col sm="12">
+                <div class="alert alert-success" v-if="isdeleted" v-on:="closeAlert()">
+                    <button aria-label="Close" class="close" data-dismiss="alert"
+                            style="line-height: 27px; margin-right: 5px;" type="button">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                    <i class="uiIconSuccess"></i>
+                    Rule successfully deleted.
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/RuleList.vue:11]: not i18ned

File: portlets/src/main/frontend/src/apps/components/rule/RuleList.vue [hunk 4]
```diff
@@ -2,39 +2,118 @@
...
+
+                                <span class="PopupTitle popupTitle">{{ this.$t('exoplatform.gamification.rule.popupdelete') }}</span>
+                            </div>
+                            <div class="PopupContent popupContent">
+                                <div class="media">
+                                    <div class="pull-left">
+                                        <i class="uiIconColorQuestion"></i>
+                                    </div>
+                                    <div class="media-body">
+                                        <p class="msg">Are you sure you want to delete this rule ?</p>
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/RuleList.vue:45]: not i18ned

File: portlets/src/main/frontend/src/apps/components/rule/RuleList.vue [hunk 5]
```diff
@@ -44,24 +123,27 @@
                                 <label class="switch">
                                     <input type="checkbox" v-model="rule.enabled">
                                     <span class="slider round"></span>
-                                    <span class="absolute-no">NO</span>
+                                    <span class="absolute-no">{{$t(`exoplatform.gamification.YES`)}}</span>
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/RuleList.vue:126]: replacing NO by YES ?

File: portlets/src/main/frontend/src/apps/components/rule/RuleList.vue [hunk 6]
```diff
@@ -44,24 +123,27 @@
...
-                            <a href="#" v-if="editedrule.id !== rule.id" v-on:click.prevent.stop="onRemove(rule.id,rule.title)" data-placement="bottom" rel="tooltip" class="actionIcon"
+
+                            <a class="actionIcon" data-placement="bottom" href="#" rel="tooltip" v-if="rule.id"
+                               v-on:click.prevent="collapseConfirm(rule)"
                                data-original-title="Supprimer" v-b-tooltip.hover title="Supprimer">
-                                <i class="uiIconDelete uiIconLightGray"></i></a>
+                                <i class="uiIconDelete uiIconLightGray"></i>
+                            </a>
                             <a href="#" v-if="editedrule.id === rule.id" v-on:click.prevent.stop="onSave(rule)" data-placement="bottom" rel="tooltip" class="actionIcon"
                                data-original-title="Edit" v-b-tooltip.hover title="Save">
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/RuleList.vue:148]: "Edit" not i18ned

File: portlets/src/main/frontend/src/apps/components/rule/SaveRuleForm.vue [hunk 7]
```diff
@@ -11,55 +14,79 @@
...
                             <b-form id="scoreInputGroup">
 
-                                <label id="scoreInputGroup" for="scoreInput" class="col-form-label pt-0">Score:</label>
+                                <label class="col-form-label pt-0" for="scoreInput" id="scoreInputGroup">{{$t(`exoplatform.gamification.score`)
+                                    }}:</label>
                                 <input id="scoreDescription" type="number" v-model="rule.score" required placeholder="Enter rule's score">
 
                                 <b-alert v-if="formErrors.score" :show="dismissCountDown" dismissible variant="danger" class="require-msg" @dismissed="dismissCountdown=0" @dismiss-count-down="countDownChanged">
-                                    Rule score is required please enter a score {{dismissCountDown}}
+                                    Rule score is required please enter a score
```
Comment (plamarque) [inline portlets/src/main/frontend/src/apps/components/rule/SaveRuleForm.vue:55]: not i18ned ?

File: services/src/main/resources/locale/addon/Gamification.properties [hunk 8]
```diff
@@ -32,50 +32,86 @@ exoplatform.gamification.gamificationinformation.domain.Teamwork=Teamwork
 exoplatform.gamification.gamificationinformation.domain.Knowledge=Knowledge
 exoplatform.gamification.gamificationinformation.domain.Feedback=Feedback
 exoplatform.gamification.gamificationinformation.myachievements=My Achievements
```
Comment (plamarque) [inline services/src/main/resources/locale/addon/Gamification.properties:34]: no need to change this file. Crowdin will do (see http://developer.exoplatform.org/#id-translations )

File: services/src/main/resources/locale/addon/Gamification_ar.properties [hunk 9]
```diff
@@ -24,7 +24,7 @@ exoplatform.gamification.leaderboard.Pieopen=\u0641\u062A\u062D \u0627\u0644\u06
 exoplatform.gamification.gamificationinformation.Total=\u0645\u062C\u0645\u0648\u0639 \u0627\u0644\u0646\u0642\u0627\u0637
```
Comment (plamarque) [inline services/src/main/resources/locale/addon/Gamification_ar.properties:24]: no need to change this file. Crowdin will do : http://developer.exoplatform.org/#id-translations

**Review by plamarque — CHANGES_REQUESTED** [review]
Spotted some strings not i18ned.

Also no need to change all .properties only Gamification_en.properties is required

see : http://developer.exoplatform.org/#id-translations

File: portlets/src/main/frontend/src/apps/components/rule/RuleList.vue (hunk 5 above)
Comment (AmineChalouati) [reply portlets/src/main/frontend/src/apps/components/rule/RuleList.vue:126]: hamadi requested  to change the default by yes

File: portlets/src/main/frontend/src/apps/components/rule/RuleList.vue (hunk 6 above)
Comment (AmineChalouati) [reply portlets/src/main/frontend/src/apps/components/rule/RuleList.vue:148]: done

File: portlets/src/main/frontend/src/apps/components/rule/SaveRuleForm.vue (hunk 7 above)
Comment (AmineChalouati) [reply portlets/src/main/frontend/src/apps/components/rule/SaveRuleForm.vue:55]: done

File: services/src/main/resources/locale/addon/Gamification_ar.properties (hunk 9 above)
Comment (AmineChalouati) [reply services/src/main/resources/locale/addon/Gamification_ar.properties:24]: ok

File: portlets/src/main/frontend/src/apps/components/rule/ManageRules.vue (hunk 1 above)
Comment (AmineChalouati) [reply portlets/src/main/frontend/src/apps/components/rule/ManageRules.vue:8]: check : https://github.com/exo-addons/gamification/pull/20

**Pull request closed without merging**